bin/PrintLowestCoveringMacroArgument/Main.cpp

```diff
@@ -0,0 +1,203 @@
+            else {
+              ss << "<unnamed> ";
+            }
+            ss << "argument number ";
+            ss << covering_argument->Index() << ' ';
+            const auto index = covering_argument->Index();
+            const auto parameter = definition->Parameters().At(index);
+            if (parameter and parameter->BeginToken()) {
+              ss << parameter->BeginToken()->Data();
+            }
+            else {
+              ss << "<unnamed>";
+            }
+          }
+          else {
+            ss << "that has no definition";
+          }
+        }
+        else {
+          ss << "that is not an expansion";
+        }
+      }
+      else {
+        ss << "that is an orphan";
+      }
+      std::cout << ss.str() << "\n";
+    }
+
+    for (const auto &child : stmt.Children()) {
+      VisitStmt(child);
+    }
+  }
+};
+
+class DeclsPrintLowestCoveringMacroArgument final : public pasta::DeclVisitor {
+public:
+  std::ostream &os;
+
+  virtual ~DeclsPrintLowestCoveringMacroArgument(void) = default;
+
+  explicit DeclsPrintLowestCoveringMacroArgument(std::ostream &os_) : os(os_) {}
+
+  void VisitDeclContext(const pasta::DeclContext &dc) {
+    for (const auto &decl : dc.AlreadyLoadedDeclarations()) {
+      Accept(decl);
+    }
+  }
+
+  void VisitTranslationUnitDecl(const pasta::TranslationUnitDecl &decl) final {
+    VisitDeclContext(decl);
+  }
+
+  void VisitVarDecl(const pasta::VarDecl &decl) final {
+    // NOTE(bpappas): Only visit global variable declarations since we visit
+    // local ones anyway inside function declaration contexts.
+    if (!decl.IsLocalVariableDeclaration()) {
+      if (auto initializer = decl.Initializer()) {
+        StmtsPrintLowestCoveringMacroArgument finder(os);
+        finder.Accept(*initializer);
+      }
+    }
+  }
+
+  void VisitFunctionDecl(const pasta::FunctionDecl &decl) final {
+    if (auto body = decl.Body()) {
+      StmtsPrintLowestCoveringMacroArgument finder(os);
+      finder.Accept(*body);
+    }
+    VisitDeclContext(decl);
+  }
+};
+
+
+static void PrintLowestCoveringMacroArgument(pasta::AST ast) {
+  DeclsPrintLowestCoveringMacroArgument finder(std::cout);
+  finder.Accept(ast.TranslationUnit());
+}
+
+static void OnSigsegv(int) {
+
+  std::cerr << "Process ID: " << getpid() << "\nPress enter: \n";
+  std::cerr.flush();
+  char x;
+  (void) read(0, &x, 1);
+}
+
+int main(int argc, char *argv[]) {
+  if (2 > argc) {
+    std::cerr << "Usage: " << argv[0] << " COMPILE_COMMAND..." << std::endl;
+    return EXIT_FAILURE;
+  }
+
+  signal(SIGSEGV, OnSigsegv);
+
+  pasta::InitPasta initializer;
+  pasta::FileManager fm(pasta::FileSystem::CreateNative());
+  auto maybe_compiler =
+    pasta::Compiler::CreateHostCompiler(fm, pasta::TargetLanguage::kCXX);
+  if (!maybe_compiler.Succeeded()) {
+    std::cerr << maybe_compiler.TakeError() << std::endl;
+    return EXIT_FAILURE;
+  }
+
+  auto maybe_cwd = pasta::FileSystem::From(maybe_compiler.Value())
```

Comment:
You created the filesystem above and passed it to `fm`. Just save the `fs` to a variable first, and used it here to get the cwd.

Reply:
This might be copypasted from my own code, so it may be my own code at fault lololol.

Reply:
Yes I just copy pasted the boiler plate from another file; I will fix this